packages/client/src/gateway-archive.ts

```diff
@@ -64,19 +69,34 @@ export namespace GatewayArchive {
      *
      * @param processArchiveData
      */
-    export function mapToGetResults(processArchiveData: IArchiveResponseBody) {
+    export function mapToGetResults(processArchiveData: IArchiveProcessResponseBody) {
         return {
             results: processArchiveData.results.votes,
             state: VochainProcessStatus[processArchiveData.process.status] || "",
             height: processArchiveData.results.envelopeHeight,
         }
     }
+
+    /**
+     * Returns the mapped data from archive to `getResults` Gateway response
+     *
+     * @param entityId
+     * @param entitiesArchiveData
+     */
+    export function mapToGetProcessList(entitiesArchiveData: IArchiveEntitiesResponseBody, entityId?: string): string[] {
+        if (entityId) {
+            return entitiesArchiveData.entities[entityId]?.map(processId => processId.processId) ?? []
+        }
+        return Object.values(entitiesArchiveData.entities)
+            .reduce((prev, curr) => prev.concat(curr), [])
+            .map(processId => processId.processId)
```

Comment:
I am not sure to understand the following block.

- Couldn't we use `entitiesArchiveData.entities.map(...)` directly, instead of `Object.values`?
- `processId.processId` too?
- Couldn't the `reduce` concat the processId's directly, instead of waiting for another `map`?

Reply:
> Couldn't we use entitiesArchiveData.entities.map(...) directly, instead of Object.values?

How can I `map` on an object? I think it's the best way to do it. If not, how would you do it?

> processId.processId too?

Is better `p => p.processId`?

> Couldn't the reduce concat the processId's directly, instead of waiting for another map?

But `curr` is an array, I should do a `map` inside the `reduce` which I think it's not so readable. Why not having first the whole array and then `map` it?